components/brave_news/rust/lib.rs

```diff
@@ -26,10 +27,55 @@ mod ffi {
     }
 
     extern "Rust" {
+        fn strip_html(subject: &str) -> String;
         fn parse_feed_bytes(source: &[u8], output: &mut FeedData) -> bool;
     }
 }
 
+// Note: This function isn't likely to be perfect, but it does guarantee that
+// there will be no HTML tags in the output (as it strips both '<' and '>' from
+// from the output).
+fn strip_html(subject: &str) -> String {
+    let mut output = String::with_capacity(subject.len());
+    let mut depth_tag = 0;
+    let mut depth_comment = 0;
+
+    for (i, c) in subject.chars().enumerate() {
+        if c == '<' {
+            let is_comment = subject.chars().skip(i).take(4).collect::<String>() == "<!--";
+            if is_comment {
+                depth_comment += 1;
+            } else if depth_comment == 0 {
+                depth_tag += 1;
+            }
+            continue;
+        }
+
+        if c == '>' {
+            if depth_comment > 0 {
+                let is_comment =
+                    i > 2 && subject.chars().skip(i - 2).take(3).collect::<String>() == "-->";
+                if is_comment {
+                    // If this was the close for a comment, reduce our comment depth.
+                    depth_comment -= 1
+                }
+            } else if depth_tag > 0 {
+                // Only reduce our tag depth if we aren't in a comment
+                depth_tag -= 1;
+            }
+            continue;
+        }
+
+        if depth_comment != 0 || depth_tag != 0 {
```

Comment:
Please use `> 0` here as you do above. It's more defensive, especially since these are signed types.

Reply:
Did you forgot to apply this change?

Reply:
Yes. Yes I did :laughing: 

Thanks!